.travis.yml

```diff
@@ -5,22 +5,38 @@ dist: bionic
 matrix:
     include:
     - compiler: clang
-      env: BUILD=cmake CPP11=OFF
+      env: BUILD=cmake CPP_STD=98 CPP11=OFF
       addons:
         apt:
           packages:
             - libc++-dev
     - compiler: clang
-      env: BUILD=cmake CPP11=ON
+      env: BUILD=cmake CPP_STD=11 CPP11=ON
+      addons:
+        apt:
+          packages:
+            - libc++-dev
+    - compiler: clang
+      env: BUILD=cmake CPP_STD=14 CPP11=ON
+      addons:
+        apt:
+          packages:
+            - libc++-dev
+    - compiler: clang
+      env: BUILD=cmake CPP_STD=17 CPP11=ON
       addons:
         apt:
           packages:
             - libc++-dev
 
     - compiler: gcc
-      env: BUILD=cmake CPP11=OFF
+      env: BUILD=cmake CPP_STD=98 CPP11=OFF
+    - compiler: gcc
+      env: BUILD=cmake CPP_STD=11 CPP11=ON
+    - compiler: gcc
+      env: BUILD=cmake CPP_STD=14 CPP11=ON
     - compiler: gcc
-      env: BUILD=cmake CPP11=ON
+      env: BUILD=cmake CPP_STD=17 CPP11=ON
```

Comment:
The CPP11=ON is a bit weird here :) Can we do something about that?

Reply:
Indeed, this is horrible. I have updated the PR to take `CMAKE_CXX_STANDARD` into account when setting the warnings. If it's set with a standard >= C++11 it'll add the warnings accordingly, making the `CPP11` option no longer necessary.